sdl2promela/sdl/model.py

```diff
@@ -1477,54 +1477,103 @@ def convert(source: ogAST.Decision) -> Action:
+        info.senderName = words[1]
+        if words[2].lower() == "to":
+            info.recipientName = words[3]
+    elif words[0].lower() == "to":
+        info.recipientName = words[1]
+
+    info.unhandled_input = False
+    return (info, parameter)
+
+
+def _parse_observer_continuous_signal_without_parameters(
+    desc: str,
+) -> Tuple[ObserverAttachmentInfo, str]:
+    info = _create_observer_info_from_signal_description(desc)
+    words = desc.split()
+    info.originalSignalName = words[1]
+    info.observerSignalName = f"obs_{words[1]}"
+
+    if words[2].lower() == "from":
+        info.senderName = words[3]
+        if words[4].lower() == "to":
+            info.recipientName = words[5]
+    elif words[2].lower() == "to":
+        info.recipientName = words[3]
+
+    info.unhandled_input = False
+    return (info, "")
+
+
+def _parse_observer_continuous_signal(desc: str) -> Tuple[ObserverAttachmentInfo, str]:
     try:
         if "(" in desc:
-            # If desc contains '(', then it contains parameter
-            # and needs to be parsed differently
-            open_position = desc.index("(")
-            close_position = desc.index(")", open_position + 1)
-            first_part = desc[0:open_position]
-            words = first_part.split()
-            info.originalSignalName = words[1]
-            info.observerSignalName = f"obs_{words[1]}"
-
-            parameter = desc[open_position + 1 : close_position]
-
-            second_part = desc[close_position + 1 :]
-            words = second_part.split()
-            if words[0].lower() == "from":
-                info.senderName = words[1]
-                if words[2].lower() == "to":
-                    info.recipientName = words[3]
-            elif words[0].lower() == "to":
-                info.recipientName = words[1]
+            return _parse_observer_continuous_signal_with_parameters(desc)
         else:
-            words = desc.split()
-            info.originalSignalName = words[1]
-            info.observerSignalName = f"obs_{words[1]}"
-
-            if words[2].lower() == "from":
-                info.senderName = words[3]
-                if words[4].lower() == "to":
-                    info.recipientName = words[5]
-            elif words[2].lower() == "to":
-                info.recipientName = words[3]
-
-        info.unhandled_input = False
-        return (info, parameter)
+            return _parse_observer_continuous_signal_without_parameters(desc)
     except ValueError:
         raise ValueError("Cannot parse continuous signal")
 
 
+def _find_signal_parameter(
+    types: Dict[str, Asn1Type], process_name: str, signal_name: str, is_input: bool
+) -> Asn1Type:
+    if is_input:
+        type_name = "{}-Event-msg-in".format(process_name.capitalize())
+    else:
+        type_name = "{}-Event-msg-out".format(process_name.captitalize())
+
+    if type_name not in types:
+        raise Exception(f"The process {process_name} cannot be found")
```

Comment:
Process?

Reply:
Yes, process. To get information about the type of parameter I need to use types generated in observer.asn, the description `input ping from controller to actuator` is not enough. Morever, if the description contains an error, it will be reported by OpenGEODE parser, so this is second check.